lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -464,6 +464,19 @@ def connect_to_machine(name, chef_server = nil)
       machine_for(machine_spec, machine_spec.reference)
     end
 
+    def stop_machine(action_handler, machine_spec, machine_options)
+      instance = instance_for(machine_spec)
+      if instance && instance.exists?
+        wait_until_machine(action_handler, machine_spec, "finish coming up so we can stop it", "up", instance) { instance.status != :pending }
+        if instance.status == :running
+          action_handler.perform_action "Stop #{machine_spec.name} (#{instance.id}) in #{aws_config.region} ..." do
+            instance.stop
+          end
+        end
+        wait_until_machine(action_handler, machine_spec, "stop", "stopped", instance) { [ :stopped, :terminated ].include?(instance.status) }
```

Comment:
Everyone learns the same lesson the hard way about `stop`ping an ephemeral instance...